api/authorization-api-1_0.md

```diff
@@ -67,187 +67,292 @@ normative:
+The core feature of the Authorization API is the Access Evaluation API, which enables a PEP to find out if a specific request can be permitted to access a specific resource. The following are non-normative examples:
+
+- Can Alice view document #123?
+- Can Alice view document #123 at 16:30 on Tuesday, June 11, 2024?
+- Can a manager print?
 
 ## API Version
 This document describes the API version 1. Any updates to this API through subsequent revisions of this document or other documents MAY augment this API, but MUST NOT modify the API described here. Augmentation MAY include additional API methods or additional parameters to existing API methods, additional authorization mechanisms, or additional optional headers in API requests. All API methods for version 1 MUST be immediately preceded by the relative URL path `/v1/`.
 
 ## Information Model
+The information model for requests and responses include the following entities: Subject, Action, Resource, Context, and Decision. These are all defined below.
 
 ### Subjects {#subjects}
-A Subject is the user or robotic principal about whom the Authorization API is being invoked. The Subject may be requesting access at the time the Authorization API is invoked, or the Subject may be of interest in a Search API call.
+A Subject is the user or robotic principal about whom the Authorization API is being invoked. The Subject may be requesting access at the time the Authorization API is invoked.
+
+A Subject is a JSON ({{RFC8259}}) object that contains any number of key-value pair attributes. However, there are a minimal number of fields that are required in order to properly resolve a Subject.
+
+`type`:
+: REQUIRED. A `string` value that specifies the type of the Subject.
+
+`id`:
+: REQUIRED. The unique identifier of the Subject, scoped to the `type`, within the scope of the PEP. If specified, the value MUST follow the format specified by the `Subject Identifiers for Security Event Tokens` specification {{RFC9493}}.
 
-A Subject is a JSON ({{RFC8259}}) object that contains any number of key-value pair attributes e.g. 
-~~~json
-"username" : "Alice",
-"department": "Sales"
+A Subject MUST minimally contain a `type` and an `id`, and MAY contain any number of additional fields.
+
+The following is a non-normative example of a subject:
+
+~~~ json
+{
+  "type": "user",
+  "id": {
+    "format" : "email",
+    "email": "alice@acmecorp.com"
+  }
+}
 ~~~
+{: #subject-example title="Example Subject"}
 
-An attribute can be single-valued or multi-valued. It can be a primitive type (string, boolean...) or a complex type such as a JSON object.
+#### Subject Identifier {#subject-identifier}
+The `id` field of a Subject MUST follow the format specified by the `Subject Identifiers for Security Event Tokens` specification {{RFC9493}}. A Subject MUST include at least one identifier, but MAY include more than one identifier if the format of the field is of type `aliases`.
```

Comment:
What I don't like with RFC 9493 is that it now forces us to specify format. It makes creating an AuthZEN request harder on the developer. It goes back to the same verbose nature of a XACML request where you had to specify datatype.

Reply:
I struggled with this too, but I thought we all agreed to use RFC 9493 because it has some pre-defined formats.

Reply:
Discussed during call - make RFC 9493 OPTIONAL